--- browser.py	2010-04-13 22:58:35.000000000 +0100
+++ browser.py.new	2010-07-14 17:10:12.000000000 +0100
@@ -34,7 +34,7 @@
 from PyQt4.QtCore import QSize, QDateTime, QVariant
 from PyQt4.QtGui import QApplication, QImage, QPainter, QRegion, QAction
 from PyQt4.QtNetwork import QNetworkCookie, QNetworkAccessManager, QNetworkReply
-from PyQt4.QtNetwork import QNetworkCookieJar, QNetworkRequest
+from PyQt4.QtNetwork import QNetworkCookieJar, QNetworkRequest, QNetworkProxy
 from PyQt4.QtWebKit import QWebPage, QWebView, QWebFrame
 
 # Debug levels
@@ -355,6 +355,11 @@
         self.webframe.load(QUrl(url))
         return self._wait_load()
 
+    def loadnetreq(self, req):
+        """Load a network request and return status (a boolean)."""
+	self.webframe.load(req)
+	return self._wait_load()
+
     def click(self, selector, wait_load=False, wait_requests=None, timeout=None):
         """
         Click any clickable element in page.
@@ -572,6 +577,45 @@
         return self.cookiesjar.setMozillaCookies(string_cookies)
 
     #}
+
+    #{ Proxies
+
+    def get_proxy(self):
+        """Return string containing the current proxy."""
+        return self.manager.proxy()
+
+    def set_proxy(self, string_proxy):
+        """Set proxy [http|socks5]://username:password@hostname:port"""
+	urlinfo = urlparse.urlparse(string_proxy)
+
+	proxy = QNetworkProxy()
+	if urlinfo.scheme == 'socks5' :
+		proxy.setType(1)
+	elif urlinfo.scheme == 'http' :
+		proxy.setType(3)
+	else : 
+		proxy.setType(2)
+		self.manager.setProxy(proxy)
+        	return self.manager.proxy()
+
+	proxy.setHostName(urlinfo.hostname)
+	proxy.setPort(urlinfo.port)
+	if urlinfo.username != None :
+		proxy.setUser(urlinfo.username)
+	else :
+		proxy.setUser('')
+
+	if urlinfo.password != None :
+		proxy.setPassword(urlinfo.password)
+	else :
+		proxy.setPassword('')
+
+	self.manager.setProxy(proxy)
+        return self.manager.proxy()
+      
+    #}
+
+
     
     #{ Download files
                 
